### PR TITLE
test: fix the reentrancy tests to call the UP instead of KeyManager

### PR DESCRIPTION
**Pull request by b00ste**
### What does this PR introduce?

In this PR we refactor the tests so that the `execute(..)` and `setData(..)` calls are made directly to the UP.

### CODE REVIEW

File: tests/LSP6KeyManager/tests/Reentrancy/SingleExecuteToSingleExecute.test.ts [hunk 1]
```diff
@@ -41,13 +41,25 @@ export const testSingleExecuteToSingleExecute = (
...
   describe("when reentering and transferring value", () => {
-    let executePayload: BytesLike;
+    let executeCalldata: {
+      operationType: number;
+      to: string;
+      value: number;
+      data: BytesLike;
+    };
     before(async () => {
-      executePayload = generateExecutePayload(
```
Comment (CJ42) [inline tests/LSP6KeyManager/tests/Reentrancy/SingleExecuteToSingleExecute.test.ts:46]: please remove this function from the reentrancyHelper and put it at the top of the file, or in the `describe` section where it is used.

You can also use the `createInterface` function directly from the factory to create instance of the main LSP interfaces and encode the function data.

image

File: tests/LSP6KeyManager/tests/Reentrancy/SingleExecuteToSingleExecuteRelayCall.test.ts [hunk 2]
```diff
@@ -38,27 +35,29 @@ export const testSingleExecuteToSingleExecuteRelayCall = (
...
-          0,
-          reentrancyContext.singleReentarncyRelayer.address,
-          0,
-          reentrantCallPayload,
-        ]
-      );
+
+    executeCalldata = {
+      operationType: 0,
+      to: reentrancyContext.singleReentarncyRelayer.address,
```
Comment (CJ42) [inline tests/LSP6KeyManager/tests/Reentrancy/SingleExecuteToSingleExecuteRelayCall.test.ts:57]:
```suggestion
      to: reentrancyContext.singleReentrancyRelayer.address,
```

**Review by CJ42 — COMMENTED** [review]
added review comments

**Comment by b00ste** [comment]
Closing the PR because we shouldn't change the tests but add new ones. 😬

**Pull request closed without merging**